arxiv/metadata/metacheck.py

```diff
@@ -120,14 +162,19 @@ def long_word_caps(s: str) -> bool:
 
 def smart_starts_with_lowercase_check(s: str) -> bool:
     """
-    Detect titles which start with a lower case char, eg "bad title"
-    but do not reject "eSpeak: A New World"
+    Detect titles which start with a lower case char, eg "bad title" or "a bad title"
+    but do not reject "eSpeak: A New World", "tuGEMM: ...", "uGMMS", or "p-Mean"
     """
 
     if len(s) < 1:
         return False
     if re.match( "[a-z]", s):
-        if re.match( "[a-z][A-Z][a-zA-Z]*: [A-Z]", s):
+        if re.match( "[a-z][a-zA-Z-]*: [A-Z]", s):
+            return False
+        elif re.match( "[a-z][A-Z][A-Z]* ", s):
+            # Matches uGMS
+            return False
+        elif re.match( "[a-z]-[A-Z]", s):
             return False
         else:
             return True

```

Comment:
I don't understand that check. As far as I read that, titles like `Fixing the Incompleteness Theorem` would  be rejected?
- it contains a lowercase `re.match("[a-z]", s)`
- it does not start with a lower case
so it returns `True` ?
Am I missing something?

Reply:
> Am I missing something?

There are no starting hats `^` in the regexes. They match anywhere within the string.

Reply:
Ahhh ...re.match only matches at the beginning of the string ... TIL